### PR TITLE
InstUI-2765 modal emotion

### PR DESCRIPTION
**Pull request by matyasf**

### CODE REVIEW

**Comment by matyasf** [comment]
note: in the dev app the images in the modal dont look good, this is a bug in the Image component, see INSTUI-2824

File: packages/ui-color-utils/src/conversions.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * Converts an color string to a hex string like `#FF0000` For possible inputs see
+ * https://github.com/bgrins/TinyColor
+ * @param {String} rgb a color string
+ * @returns {String} a hex string like `#FF0000`
+ */
+function color2hex(rgb) {
+  return Color(rgb).toHex()
+}
+
+export { color2hex }
```
Comment (matyasf) [inline packages/ui-color-utils/src/conversions.js:40]: Utiliy method used in tests

**Comment by ke1k0** [comment]
The refact branch is merged into next, so please make the modifications to match the new componentTheme handling: https://github.com/instructure/instructure-ui/pull/316.

File: packages/ui-modal/src/Modal/__tests__/Modal.test.js [hunk 2]
```diff
@@ -55,25 +57,33 @@ describe('<Modal />', async () => {
...
+          <Modal
+            open
+            size="small"
+            label="Modal Dialog"
+            shouldReturnFocus={false}
+          >
+            <Modal.Body>Foo Bar Baz</Modal.Body>
+          </Modal>
+        </EmotionThemeProvider>
+      </EmotionThemeProvider>
```
Comment (ke1k0) [inline packages/ui-modal/src/Modal/__tests__/Modal.test.js:86]: The second EmotionThemeProvider is not needed, you can simply add the override directly to the component on the `themeOverride` key.

```suggestion
      <EmotionThemeProvider theme={canvas}>
          <Modal
            open
            size="small"
            label="Modal Dialog"
            shouldReturnFocus={false}
            themeOverride={{ smallMaxWidth: '10em' }}
          >
            <Modal.Body>Foo Bar Baz</Modal.Body>
          </Modal>
      </EmotionThemeProvider>
```

File: packages/ui-modal/src/Modal/__tests__/Modal.test.js [hunk 3]
```diff
@@ -27,7 +27,9 @@ import { expect, mount, stub, wait, within } from '@instructure/ui-test-utils'
 
 import { Modal } from '../index'
 import { ModalLocator } from '../ModalLocator'
-import styles from '../styles.css'
+import { EmotionThemeProvider } from '@instructure/emotion'
+import generateComponentTheme from '../theme'
```
Comment (ke1k0) [inline packages/ui-modal/src/Modal/__tests__/Modal.test.js:31]: Unused import.

File: packages/ui-modal/src/Modal/__tests__/Modal.test.js [hunk 4]
```diff
@@ -97,25 +107,31 @@ describe('<Modal />', async () => {
...
-          label="Modal Dialog"
-          shouldReturnFocus={false}
-          constrain="parent"
-          theme={{
-            textColor: 'rgb(0, 0, 0)'
-          }}
-        >
-          <Modal.Body>Foo Bar Baz</Modal.Body>
-        </Modal>
+        <EmotionThemeProvider theme={themeOverride}>
```
Comment (ke1k0) [inline packages/ui-modal/src/Modal/__tests__/Modal.test.js:125]: Same here, you can add the override directly to the component.

File: packages/ui-modal/src/Modal/styles.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+ */
+const generateStyle = (theme, themeOverride, props, state) => {
+  // get the theme variables object for the component
+  const componentTheme = generateComponentTheme(theme, themeOverride)
+  const { size, variant, overflow, constrain } = props
+
+  const commonSizeStyleExceptForFullscreen = {
+    maxWidth: '95%',
+    maxHeight: '95%',
+    ...(overflow && {
```
Comment (ke1k0) [inline packages/ui-modal/src/Modal/styles.js:42]: Th original class was applied when `[styles['overflow--fit']]: this.props.overflow === 'fit',`, so I think it should be:
```js
...(overflow === 'fit' && {
```

File: packages/ui-modal/src/Modal/ModalBody/index.js [hunk 6]
```diff
@@ -22,35 +22,38 @@
...
+import { ThemeablePropTypes } from '@instructure/ui-themeable'
 import { testable } from '@instructure/ui-testable'
-import { error } from '@instructure/console/macro'
-import { isIE11 } from '@instructure/ui-utils'
-
-import styles from './styles.css'
-import theme from './theme'
+import { withStyle, jsx } from '@instructure/emotion'
+import generateStyle from './styles'
+import { omitProps } from '@instructure/ui-react-utils'
```
[the remark this answers is not in the document]
Comment (matyasf) [reply packages/ui-modal/src/Modal/ModalBody/index.js:34]: hmm we could check if our linter can organize imports. If not, then we should not bother with it IMO

File: packages/ui-modal/src/Modal/styles.js (hunk 5 above)
Comment (matyasf) [reply packages/ui-modal/src/Modal/styles.js:42]: good catch :)

**Review by ke1k0 — APPROVED** [review]

**Review by HerrTopi — APPROVED** [review]

**Pull request merged**